api/v1alpha1/workstatus_types.go

```diff
@@ -31,13 +31,21 @@ type WorkStatus struct {
 
 	Spec   WorkStatusSpec `json:"spec,omitempty"`
 	Status RawStatus      `json:"status,omitempty"`
+	StatusDetails StatusDetails `json:"statusDetails, omitempty"`
 }
 
 // Workstatus spec
 type WorkStatusSpec struct {
 	SourceRef SourceRef `json:"sourceRef,omitempty"`
 }
 
+// StatusDetails spec
```

Comment:
FYI, I would be a little more expansive in the comment here. Explain that this is data about the downsync propagation.